kafka/consumer/fetcher.py

```diff
@@ -873,29 +824,29 @@ def _handle_fetch_response(self, request, send_time, response):
                                   position)
                         continue
 
+                    if not records.has_next() and records.size_in_bytes() > 0:
+                        # we did not read a single message from a non-empty
+                        # buffer because that message's size is larger than
+                        # fetch size, in this case record this exception
+                        self._record_too_large_partitions[tp] = fetch_offset
```

Comment:
According to the Java consumer docs, this is actually now a soft-limit so that the consumer can make progress. That change is likely outside the scope of this PR, so I filed https://github.com/dpkp/kafka-python/issues/1232 to track that.